### DISCUSSION TITLE
Decide which blockchain to use

### DISCUSSION THREAD

**Discussion by ParkerH27** [PSDTools/app#40] [General]
## Blockchains

Chat-GPT recommended:
- [BigChainDB](https://www.bigchaindb.com/features/)
- [Quorum Blockchain](https://www.kaleido.io/blockchain-platform/quorum)

## ChatGPT

<details><summary><h3>BigChainDB</h3></summary>

#### Prompt

> How would I set up BigChainDB as a server in Python with a Dart/Flutter client?

#### ChatGPT:

> To set up BigchainDB as a server using Python and create a Dart/Flutter client to interact with it, you can follow these steps:
> 
> ##### Setting up BigchainDB Server:
> 
> ###### Install BigchainDB:
> Begin by installing BigchainDB on your server. Refer to the official BigchainDB documentation for detailed instructions based on your operating system: <https://docs.bigchaindb.com/en/latest/installing-server.html>
> 
> ###### Configure BigchainDB:
> After installation, configure the BigchainDB server by modifying the configuration file (<bigchaindb.conf>). Adjust the settings as per your requirements, such as the database backend, API endpoints, and network parameters. Refer to the official documentation for configuration details: <https://docs.bigchaindb.com/projects/server/en/latest/drivers-clients/http-client-server-api.html#configuration>
> 
> ###### Start BigchainDB Server:
> Once configured, start the BigchainDB server. You can typically do this by running a command such as `bigchaindb start`. Verify that the server is running correctly and accessible through the configured API endpoints.
> 
> ##### Creating a Dart/Flutter Client:
> 
> ###### Set Up Flutter Project:
> Create a new Flutter project or use an existing one. Ensure you have the necessary Dart and Flutter SDKs installed on your development machine.
> 
> ###### Add Dependencies:
> Open the pubspec.yaml file in your Flutter project and add the required dependencies to interact with the BigchainDB server. You can use the `package:http` for making HTTP requests and `dart:convert` for JSON parsing. Here's an example of adding dependencies in a `pubspec.yaml`:
> 
> ```yaml
> dependencies:
>   http: ^0.13.3
>   dart:convert: ^3.0.1
> ```
>
> ###### Implement API Calls: 
> In your Dart/Flutter project, create the necessary code to interact with the BigchainDB server through HTTP requests. Use the `package:http` to make GET, POST, PUT, or DELETE requests to the appropriate endpoints provided by the BigchainDB server API.
> 
> For example, you can use the `package:http` to send a GET request to retrieve a transaction based on its ID:
> 
> ```dart
> import 'package:http/http.dart' as http;
> import 'dart:convert';
> 
> Future<Map<String, dynamic>> getTransactionById(String id) async {
>   final response = await http.get(Uri.parse('http://your-bigchaindb-server/api/v1/transactions/$id'));
>   if (response.statusCode == 200) {
>     return json.decode(response.body);
>   } else {
>     throw Exception('Failed to retrieve transaction');
>   }
> }
> ```
>
> ###### Integrate API Calls in the UI:
> Use the API calls you implemented in your Flutter UI to display and interact with the data from the BigchainDB server. Design your UI screens to allow users to perform actions such as creating transactions, querying transactions, or updating asset information.
> 
> ###### Test and Run:
> Test your Dart/Flutter client by running the app on an emulator, simulator, or physical device. Verify that the app successfully communicates with the BigchainDB server and performs the desired operations.
> 
> Please note that the example provided is a basic illustration of interacting with the BigchainDB server using Dart/Flutter. Depending on your specific requirements, you may need to implement additional API calls and handle authentication or other security considerations. Refer to the BigchainDB API documentation for more details on available endpoints and how to interact with the server: <https://docs.bigchaindb.com/projects/server/en/latest/drivers-clients/http-client-server-api.html>
> 
> Remember to adapt the code and configurations to your specific server and API endpoints.
> 

</details>

<details><summary><h3>Quorum</h3></summary>

#### Prompt

> How would I set up Quorum as a server in Python with a Dart/Flutter client?

#### ChatGPT:

> Setting up Quorum as a server in Python and creating a Dart/Flutter client involves several steps. Here's a high-level overview of the process:
> 
> ##### Set Up Quorum Server:
> 
> ###### Install Quorum:
> Follow the official Quorum documentation to install and configure Quorum on your server. This typically involves setting up a network of nodes and configuring the consensus mechanism (e.g., Raft, Istanbul BFT, etc.).
> ###### Configure the network:
> Define the network configuration, including the number of nodes, their IP addresses, ports, and the consensus algorithm you want to use.
> ###### Deploy contracts:
> Write and deploy the necessary smart contracts to manage your fake currency. You can use Solidity to write the contracts, and the Quorum-specific toolchain (e.g., Truffle with Quorum plugin) for deployment.
>
> ##### Develop the Python Server:
> 
> ###### Choose a Python web framework:
> Select a suitable Python web framework like Flask or Django to develop the server-side application. These frameworks provide the necessary tools for handling HTTP requests, managing routes, and interacting with the Quorum network.
> ###### Connect to Quorum:
> Use a Quorum-specific Python library like Web3.py to establish a connection between your Python server and the Quorum network. `Web3.py` provides APIs to interact with smart contracts, send transactions, and query the blockchain.
> ###### Implement the server-side logic:
> Write the necessary Python code to handle requests from the Flutter client, such as creating transactions, querying balances, and interacting with the smart contracts deployed on Quorum.
>
> ##### Develop the Dart/Flutter Client:
> 
> ###### Set up Dart/Flutter:
> Install the Dart SDK and Flutter framework on your development machine.
> ###### Create a Flutter project:
> Use the Flutter CLI to create a new Flutter project.
> ###### Implement the user interface:
> Design and develop the UI of your app using Flutter widgets. This includes screens for creating transactions, displaying balances, and other desired functionality.
> ###### Implement the client-side logic:
> Write Dart code to handle interactions with the Python server. Use HTTP libraries like Dio or the Flutter-specific `package:http` to send requests to the server and process the responses.
> ###### Test and Debug:
> Test the server-side application and the Flutter client to ensure they function correctly. Perform unit tests and integration tests to identify and fix any issues or bugs.
> Debug any issues that arise during testing and address them accordingly.
> ###### Deploy and Maintain:
> Deploy the Python server to a hosting platform or your own server infrastructure. Ensure that the server is accessible and secured properly.
> Package and distribute the Flutter app for deployment on Android and iOS devices.
> Regularly maintain and update both the server-side application and the Flutter client to improve functionality, security, and compatibility with new versions of Quorum and Flutter.
>
> It's important to note that the steps mentioned above provide a general guideline, and the specific implementation details may vary depending on your project requirements and the versions of Quorum, Python, and Flutter you are using. Refer to the official documentation and relevant resources for more detailed instructions and examples.

</details>

**Comment by ParkerH27** [comment]
Blockchains....
Chat-GPT recommended:
- [BigChainDB](https://www.bigchaindb.com/features/)
- [Quorum Blockchain](https://www.geeksforgeeks.org/quorum-blockchain/)

**Comment by ParkerH27** [comment]
Bigchain
how would I set up BigChainDB as a server in python with a dart/flutter client

ChatGPT:
To set up BigchainDB as a server using Python and create a Dart/Flutter client to interact with it, you can follow these steps:

Setting up BigchainDB Server:

Install BigchainDB: Begin by installing BigchainDB on your server. Refer to the official BigchainDB documentation for detailed instructions based on your operating system: https://docs.bigchaindb.com/en/latest/installing-server.html

Configure BigchainDB: After installation, configure the BigchainDB server by modifying the configuration file (bigchaindb.conf). Adjust the settings as per your requirements, such as the database backend, API endpoints, and network parameters. Refer to the official documentation for configuration details: https://docs.bigchaindb.com/projects/server/en/latest/drivers-clients/http-client-server-api.html#configuration

Start BigchainDB Server: Once configured, start the BigchainDB server. You can typically do this by running a command such as bigchaindb start. Verify that the server is running correctly and accessible through the configured API endpoints.

Creating a Dart/Flutter Client:

Set Up Flutter Project: Create a new Flutter project or use an existing one. Ensure you have the necessary Dart and Flutter SDKs installed on your development machine.

Add Dependencies: Open the pubspec.yaml file in your Flutter project and add the required dependencies to interact with the BigchainDB server. You can use the http package for making HTTP requests and dart:convert for JSON parsing. Here's an example of adding dependencies in pubspec.yaml:

yaml
Copy code
dependencies:
  http: ^0.13.3
  dart:convert: ^3.0.1
Implement API Calls: In your Dart/Flutter project, create the necessary code to interact with the BigchainDB server through HTTP requests. Use the http package to make GET, POST, PUT, or DELETE requests to the appropriate endpoints provided by the BigchainDB server API.

For example, you can use the http package to send a GET request to retrieve a transaction based on its ID:

dart
Copy code
import 'package:http/http.dart' as http;
import 'dart:convert';

Future<Map<String, dynamic>> getTransactionById(String id) async {
  final response = await http.get(Uri.parse('http://your-bigchaindb-server/api/v1/transactions/$id'));
  if (response.statusCode == 200) {
    return json.decode(response.body);
  } else {
    throw Exception('Failed to retrieve transaction');
  }
}
Integrate API Calls in the UI: Use the API calls you implemented in your Flutter UI to display and interact with the data from the BigchainDB server. Design your UI screens to allow users to perform actions such as creating transactions, querying transactions, or updating asset information.

Test and Run: Test your Dart/Flutter client by running the app on an emulator, simulator, or physical device. Verify that the app successfully communicates with the BigchainDB server and performs the desired operations.

Please note that the example provided is a basic illustration of interacting with the BigchainDB server using Dart/Flutter. Depending on your specific requirements, you may need to implement additional API calls and handle authentication or other security considerations. Refer to the BigchainDB API documentation for more details on available endpoints and how to interact with the server: https://docs.bigchaindb.com/projects/server/en/latest/drivers-clients/http-client-server-api.html

Remember to adapt the code and configurations to your specific server and API endpoints.

**Comment by ParkerH27** [comment]
Quorum
how would I set up quorum as a server in python with a dart/flutter client
ChatGPT:
Setting up Quorum as a server in Python and creating a Dart/Flutter client involves several steps. Here's a high-level overview of the process:

Set Up Quorum Server:

Install Quorum: Follow the official Quorum documentation to install and configure Quorum on your server. This typically involves setting up a network of nodes and configuring the consensus mechanism (e.g., Raft, Istanbul BFT, etc.).
Configure the network: Define the network configuration, including the number of nodes, their IP addresses, ports, and the consensus algorithm you want to use.
Deploy contracts: Write and deploy the necessary smart contracts to manage your fake currency. You can use Solidity to write the contracts, and the Quorum-specific toolchain (e.g., Truffle with Quorum plugin) for deployment.
Develop the Python Server:

Choose a Python web framework: Select a suitable Python web framework like Flask or Django to develop the server-side application. These frameworks provide the necessary tools for handling HTTP requests, managing routes, and interacting with the Quorum network.
Connect to Quorum: Use a Quorum-specific Python library like Web3.py to establish a connection between your Python server and the Quorum network. Web3.py provides APIs to interact with smart contracts, send transactions, and query the blockchain.
Implement the server-side logic: Write the necessary Python code to handle requests from the Flutter client, such as creating transactions, querying balances, and interacting with the smart contracts deployed on Quorum.
Develop the Dart/Flutter Client:

Set up Dart/Flutter: Install the Dart SDK and Flutter framework on your development machine.
Create a Flutter project: Use the Flutter CLI to create a new Flutter project.
Implement the user interface: Design and develop the UI of your app using Flutter widgets. This includes screens for creating transactions, displaying balances, and other desired functionality.
Implement the client-side logic: Write Dart code to handle interactions with the Python server. Use HTTP libraries like Dio or the Flutter-specific package http to send requests to the server and process the responses.
Test and Debug:

Test the server-side application and the Flutter client to ensure they function correctly. Perform unit tests and integration tests to identify and fix any issues or bugs.
Debug any issues that arise during testing and address them accordingly.
Deploy and Maintain:

Deploy the Python server to a hosting platform or your own server infrastructure. Ensure that the server is accessible and secured properly.
Package and distribute the Flutter app for deployment on Android and iOS devices.
Regularly maintain and update both the server-side application and the Flutter client to improve functionality, security, and compatibility with new versions of Quorum and Flutter.
It's important to note that the steps mentioned above provide a general guideline, and the specific implementation details may vary depending on your project requirements and the versions of Quorum, Python, and Flutter you are using. Refer to the official documentation and relevant resources for more detailed instructions and examples.

**Comment by lishaduck** [comment]
Seems like BigChain is a whole lot more complicated, but Quorum is like...real. So, I would opt for Quorum, but it might be trickier in the long run. I suppose that's good though, and it's more powerful.

**Reply by lishaduck** [reply]
Update: Quorum it is!

**Reply by lishaduck** [reply]
More looking...and...maybe? Consys gave QBS to Kaleido, and Kaleido sandboxes only last for 2 months.

**Reply by lishaduck** [reply]
However, we can still self-host Quorum on Azure, but Azure ain't free.

**Comment by StephanieBrooks** [comment]
Discover the power of blockchain with EOSedu! Whether you're a beginner or an experienced developer, our platform offers in-depth courses on EOSIO technology. From smart contract development to decentralized app creation, our step-by-step tutorials and real-world projects will elevate your skills. Join our vibrant community and unlock your potential in the world of blockchain. Enroll now: [https://eosedu.com/](https://eosedu.com/)
